resources/services/sagemaker/training_jobs.go

```diff
@@ -562,34 +566,50 @@ func SagemakerTrainingJobs() *schema.Table {
 //                                               Table Resolver Functions
 // ====================================================================================================================
 
+func fetchTrainingJobDefinition(ctx context.Context, res chan<- interface{}, svc client.SageMakerClient, region string, n types.TrainingJobSummary) error {
+	config := sagemaker.DescribeTrainingJobInput{
+		TrainingJobName: n.TrainingJobName,
+	}
+	response, err := svc.DescribeTrainingJob(ctx, &config, func(options *sagemaker.Options) {
+		options.Region = region
+	})
+	if err != nil {
+		return diag.WrapError(err)
+	}
+
+	res <- response
+	return nil
+}
+
 func fetchSagemakerTrainingJobs(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- interface{}) error {
 	c := meta.(*client.Client)
 	svc := c.Services().SageMaker
 	config := sagemaker.ListTrainingJobsInput{}
+	var sem = semaphore.NewWeighted(int64(MAX_GOROUTINES))
+
 	for {
 		response, err := svc.ListTrainingJobs(ctx, &config, func(options *sagemaker.Options) {
 			options.Region = c.Region
 		})
 		if err != nil {
 			return diag.WrapError(err)
 		}
-
-		// get more details about the notebook instance
-		for _, n := range response.TrainingJobSummaries {
-
-			config := sagemaker.DescribeTrainingJobInput{
-				TrainingJobName: n.TrainingJobName,
-			}
-			response, err := svc.DescribeTrainingJob(ctx, &config, func(options *sagemaker.Options) {
-				options.Region = c.Region
-			})
-			if err != nil {
+		errs, ctx := errgroup.WithContext(ctx)
+		for _, d := range response.TrainingJobSummaries {
```

Comment:
I would List all jobs first, finish the pagination, then do the errgroupy things. Otherwise if the 9th training job summary were to hang when all others are finished, we don't start processing the next page.

Unless you can have huge number of jobs, in that case a waitgroup/channel approach could be better (or separate resource/resolver)

Reply:
I didn't think of this edge case before, or didn't notice we were doing pagination in the previous resources. Maybe merge this for now and work on something better for later, possibly an SDK change.